src/App.tsx

```diff
@@ -1,27 +1,95 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
+import { Loader } from './Loader';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum SortBy {
+  ALL = 'all-button',
+  FIRST5 = 'first-five-button',
+  RED = 'red-button',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+type ButtonNames = {
+  [key in SortBy]: string;
+};
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const buttonNames: ButtonNames = {
+  [SortBy.ALL]: 'Load all goods',
+  [SortBy.FIRST5]: 'Load 5 first goods',
+  [SortBy.RED]: 'Load red goods',
+};
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sortBy, setSortBy] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [loading, setLoading] = useState(false);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const getGoods = async (promise: Promise<Good[]>): Promise<void> => {
+    setLoading(true);
+    setErrorText('');
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    try {
+      const goodsFromServer = await promise;
+
+      setGoods(goodsFromServer);
+    } catch (error) {
+      setErrorText(String(error));
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleButtonClick = (sort: SortBy) => {
+    if (sortBy === sort) {
+      return;
+    }
+
+    setSortBy(sort);
+
+    switch (sort) {
+      case SortBy.ALL:
+        getGoods(goodsAPI.getAll());
```

Comment:
```suggestion
        getGoods(goodsAPI.getAll);
```

Pass a reference, not a result

Reply:
![Знімок екрана (102)](https://user-images.githubusercontent.com/116962817/231168046-8b45f475-aa08-4e48-a25c-3be59028b824.png)
cannot do this